### PR TITLE
Bundled C core with Ruby library

### LINKED ISSUE #4398: Bundle C core with Ruby
Ruby needs to have a single install step: `gem install grpc`. This means no dependency on the gRPC homebrew package or the `libgrpc-dev` debian package.

### PR DESCRIPTION
**Pull request by murgatroid99**
Note: this currently uses the system OpenSSL installation. When we start distributing BoringSSL, we should switch to using that.

Fixes #4398 and #4515.

### CODE REVIEW

File: grpc.gemspec [hunk 1]
```diff
@@ -16,17 +16,20 @@ Gem::Specification.new do |s|
...
-  s.files += Dir.glob('lib/**/*')
-  s.files += Dir.glob('pb/**/*')
-  s.test_files = Dir.glob('spec/**/*')
+  s.files = %w( Rakefile Makefile )
+  s.files += Dir.glob('src/ruby/bin/**/*')
+  s.files += Dir.glob('src/ruby/ext/**/*')
+  s.files += Dir.glob('src/ruby/lib/**/*')
+  s.files += Dir.glob('src/ruby/pb/**/*')
+  s.files += Dir.glob('include/grpc/**/*')
+  s.files += Dir.glob('src/core/**/*')
```
Comment (ctiller) [inline grpc.gemspec]: Can we template this?

There have been cases (and likely will be cases again) where we have code in src/core that we don't necessarily want to bundle up for distribution, but nor do we want to delete it.

**Comment by jtattermusch** [comment]
Besides craig's comment, LGTM.

(Please ignore portability tests, they should be passing soon but aren't yet).

File: grpc.gemspec (hunk 1 above)
Comment (murgatroid99) [inline grpc.gemspec]: done

**Comment by jtattermusch** [comment]
LGTM, fine to merge once interop tests pass.

**Comment by murgatroid99** [comment]
The only interop failure is the C++ -> Java test that keeps failing.

**Pull request merged**